app/src/main/java/com/impulse/afterdarrk/Actions/DirectionButton.java

```diff
@@ -0,0 +1,64 @@
+package com.impulse.afterdarrk.Actions;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Rect;
+import android.view.MotionEvent;
+import android.view.View;
+
+import com.impulse.afterdarrk.DisplayObj;
+import com.impulse.afterdarrk.Main;
+import com.impulse.afterdarrk.Player;
+import com.impulse.afterdarrk.R;
+import com.impulse.afterdarrk.Utils.CartesianCoords;
+
+public class DirectionButton extends DisplayObj {
+    private final Player player;
+    private final CartesianCoords position;
+    private final CartesianCoords size;
+    private final DirectionType directionType;
+    private final Bitmap bitmap;
+
+    public DirectionButton(Player player, CartesianCoords position, CartesianCoords size, DirectionType direction) {
+        this.player = player;
+        this.position = position;
+        this.directionType = direction;
+        this.size = size;
+        switch (directionType) {
+            case LEFT:
+                bitmap = Bitmap.createScaledBitmap(BitmapFactory.decodeResource(Main.context.getResources(), R.drawable.left), (int) Math.round(size.getX()), (int) Math.round(size.getY()), true);
```

Comment:
The only thing that is different in case LEFT and case RIGHT is R.drawable.X. Can we refactor out the rest of the calculations to above the switch to reduce the line length